metricflow/specs/patterns/entity_link_pattern.py

```diff
@@ -0,0 +1,174 @@
+from __future__ import annotations
```

Comment:
Expanded the commit message, but also left the link in there as I thought it would still be useful. It's now:

```
Add SpecPattern classes.

These pattern classes are used to model user inputs when group-by items are
specified through the query interface or specified in a filter. These patterns
allow for ambiguous user inputs of group-by items e.g. a time dimension with
an unknown grain. For that case, the ambiguous input makes it easier for the
user to author queries as figuring out the time grain requires inspection of
the configs. For more details, please see:

https://github.com/dbt-labs/metricflow/issues/887
```